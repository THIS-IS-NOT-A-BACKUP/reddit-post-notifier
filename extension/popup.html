<!DOCTYPE html>

<html>

<head>
  <meta charset="utf-8">
  <link rel="stylesheet" href="styles/popup.css">
  <script src="bundles/popup.js"></script>
</head>

<body>
  <!-- Check icon -->
  <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 16 16" style="display: none;">
    <path id="check-mark-path"
      d="M6 14a1 1 0 0 1-.707-.293l-3-3a1 1 0 0 1 1.414-1.414l2.157 2.157 6.316-9.023a1 1 0 0 1 1.639 1.146l-7 10a1 1 0 0 1-.732.427A.863.863 0 0 1 6 14z">
    </path>
  </svg>

  <header>
    <button class="arrow-left"><img src="images/arrowhead-left.svg"></img></button>
    <a class="subreddit-name" href="reddit.com"></a>
    <button class="options"><img src="images/settings.svg"></img></button>
    <a class="mail">
      <span class="unread-number"></span>
      <svg xmlns="http://www.w3.org/2000/svg" width="16" height="16" viewBox="0 0 16 16">
        <path
          d="M13 2H3C1.3 2 0 3.3 0 5v6c0 1.7 1.3 3 3 3h10c1.7 0 3-1.3 3-3V5c0-1.7-1.3-3-3-3zm0 2c.1 0 .2 0 .3.1h-.1L8 7.9 2.8 4.1h-.1c.1-.1.2-.1.3-.1h10zm1 7c0 .6-.4 1-1 1H3c-.6 0-1-.4-1-1V4.8c0 .1.1.1.2.1l5.5 4c.1.1.2.1.3.1.1 0 .2 0 .3-.1l5.5-4c.1 0 .1-.1.1-.2.1.1.1.2.1.3v6z">
        </path>
      </svg>
    </a>
  </header>

  <main>
    <!-- List of subreddits and queries -->
    <template id="query-list-tmp">
      <ul class="query-list">
      </ul>
    </template>
    <template id="query-list-row-tmp">
      <li tabindex="1">
        <button class="check-mark">
          <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 16 16">
            <use href="#check-mark-path">
          </svg>
        </button>
        <div class="query-name"></div>
        <button class="arrow-right"><img src="images/arrowhead-right.svg"></img></button>
      </li>
    </template>

    <template id="no-posts">
      <div class="no-posts">No unread posts</div>
    </template>

    <!-- List of posts  -->
    <template id="post-list-tmp">
      <ul class="post-list"></ul>
    </template>
    <template id="post-list-row-tmp">
      <li tabindex="1">
        <button class="check-mark">
          <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 16 16">
            <use href="#check-mark-path">
          </svg>
        </button>
        <a href="#"></a>
      </li>
    </template>

  </main>

  <footer>
    <button class="footer-button">Mark all as read</button>
  </footer>

  <div id="preview"></div>

</body>

</html>
