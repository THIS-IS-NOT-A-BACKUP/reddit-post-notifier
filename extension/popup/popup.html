<!DOCTYPE html>

<html>

<head>
  <meta charset="utf-8">
  <link rel="stylesheet" href="../styles/popup.css">
  <script src="../bundles/popup.js"></script>
</head>

<body>
  <div>Authorized!</div>
</body>

</html>
