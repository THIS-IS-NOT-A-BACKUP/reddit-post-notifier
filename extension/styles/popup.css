* {
    box-sizing: border-box;
}

html {
    font-size: 62.5%; /* ==10px if default font size is 16px */
    font-family: "PT Sans", "Trebuchet MS", Arial, sans-serif;
}

body {
    min-width: 150px;
    max-width: 800px;
    max-height: 600px;
}
