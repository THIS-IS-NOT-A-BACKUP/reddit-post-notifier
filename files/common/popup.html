<!DOCTYPE html>

<html>
    <head>
        <meta charset="utf-8" />
        <script src="bundles/popup.js" defer type="module"></script>
    </head>

    <body></body>
</html>
