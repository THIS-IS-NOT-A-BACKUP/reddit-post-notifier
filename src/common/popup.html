<!DOCTYPE html>

<html>
    <head>
        <meta charset="utf-8" />
        <link rel="stylesheet" href="styles/nprogress.css" />
        <link rel="stylesheet" href="styles/colors.css" />
        <script src="browser-polyfill.js"></script>
        <script defer src="bundles/popup.js"></script>
    </head>

    <body></body>
</html>
