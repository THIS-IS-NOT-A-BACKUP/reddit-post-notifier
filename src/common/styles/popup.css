@import './colors.css';

* {
    box-sizing: border-box;
}

body {
    color: var(--font-color);
    margin: 0;
    min-width: 200px;
    max-width: 500px;
    display: flex;
    flex-direction: column;
    background-color: var(--main-bg-color);
    overflow: hidden auto;
}

svg {
    fill: var(--font-color);
    color: var(--font-color);
}

a {
    text-decoration: none;
    color: var(--link-color);
}
