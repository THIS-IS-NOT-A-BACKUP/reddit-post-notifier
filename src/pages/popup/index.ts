import Popup from '@/pages/popup/components/Popup.svelte';

// eslint-disable-next-line @typescript-eslint/no-non-null-assertion
const popup = new Popup({ target: document.getElementById('app')! });

export default popup;
