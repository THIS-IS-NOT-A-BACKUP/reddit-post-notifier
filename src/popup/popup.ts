import Popup from './components/Popup.svelte';

const app = new Popup({ target: document.body });

export default app;
