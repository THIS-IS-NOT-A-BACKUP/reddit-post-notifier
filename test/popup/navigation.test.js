import './mocks/document-body.mock';
import './mocks/browser.mock';
import './mocks/storage.mock';
import renderQueryListBlock from './mocks/render-query-list.mock';
import renderPostListBlock from './mocks/render-post-list.mock';
import nav from '../../scripts/popup/navigation';
import updateHeader from '../../scripts/popup/updateHeader';
import updateFooter from '../../scripts/popup/updateFooter';
import getElements from '../../scripts/popup/elements';
import { data } from '../../scripts/popup/data';

jest.mock('../../scripts/popup/updateFooter.js', () => jest.fn());

jest.mock('../../scripts/popup/updateHeader.js', () => jest.fn());

const elements = getElements();

describe('navigation', () => {
    afterEach(() => jest.clearAllMocks());

    test('should navigate to queriesList', async () => {
        await nav.navigate(nav.locations.queriesList, { forceUpdate: true });

        expect(nav.locations.current).toBe(nav.locations.queriesList);
        expect(updateHeader).toHaveBeenCalledWith(nav.locations.queriesList);
        expect(updateFooter).toHaveBeenCalledWith(nav);
        expect(document.body.style.minHeight).toBe('');
        expect(document.body.style.minWidth).toBe('');
        expect(renderQueryListBlock).toHaveBeenCalledTimes(1);
        expect(elements.mainContainer).toMatchSnapshot();
    });

    test('should navigate to the list of subreddit\'s posts', async () => {
        const id = 'subreddit';
        const params = { id, type: 'r' };

        await nav.navigate(nav.locations.postList, params);

        expect(nav.locations.current).toBe(nav.locations.postList);
        expect(document.body.style.minHeight).toBe('300px');
        expect(document.body.style.minWidth).toBe('400px');
        expect(updateHeader).toHaveBeenCalledWith(nav.locations.postList, {
            name: `r/${id}/new`,
            href: `https://reddit.com/r/${id}/new`,
        });
        expect(updateFooter).toHaveBeenCalledWith(nav, {
            subreddit: id,
        });
        expect(renderPostListBlock).toHaveBeenCalledWith({ subreddit: id });
    });

    test('should navigate to the list of the subreddit search queries\'s posts', async () => {
        const { id } = data.watchQueries[0];
        const params = { id, type: 's' };

        await nav.navigate(nav.locations.postList, params);

        expect(nav.locations.current).toBe(nav.locations.postList);
        expect(document.body.style.minHeight).toBe('300px');
        expect(document.body.style.minWidth).toBe('400px');

        expect(updateHeader.mock).toMatchSnapshot();
        expect(updateFooter).toHaveBeenCalledWith(nav, { searchId: id });
        expect(renderPostListBlock).toHaveBeenCalledWith({ search: id });
        expect(elements.mainContainer).toMatchSnapshot();
    });

    test('should navigate to the list of the reddit search queries\'s posts', async () => {
        const { id } = data.watchQueries[0];
        const params = { id, type: 's' };
        jest.spyOn(data.watchQueries, 'find').mockImplementationOnce(() => ({
            name: 'Query name',
            query: 'user_query',
            subreddit: '',
        }));
        await nav.navigate(nav.locations.postList, params);
        expect(updateHeader.mock).toMatchSnapshot();
        expect(updateFooter).toHaveBeenCalledWith(nav, { searchId: id });
        expect(renderPostListBlock).toHaveBeenCalledWith({ search: id });
    });
});
