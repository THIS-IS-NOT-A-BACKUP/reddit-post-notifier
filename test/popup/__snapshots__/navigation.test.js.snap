// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`navigation should navigate to queriesList 1`] = `
<main>
  <ul
    class="query-list"
  />
</main>
`;

exports[`navigation should navigate to the list of the reddit search queries's posts 1`] = `
Array [
  Array [
    "postList",
    Object {
      "href": "https://reddit.com/search?q=user_query&sort=new",
      "name": "Query name",
    },
  ],
]
`;

exports[`navigation should navigate to the list of the subreddit search queries's posts 1`] = `
Array [
  Array [
    "postList",
    Object {
      "href": "https://reddit.com/r/javascript/search?q=react&sort=new&restrict_sr=on",
      "name": "react in r/javascript",
    },
  ],
]
`;

exports[`navigation should navigate to the list of the subreddit search queries's posts 2`] = `
<main>
  <ul
    class="post-list"
  />
</main>
`;
